waiter/src/waiter/cookie_support.clj

```diff
@@ -67,11 +55,20 @@
     (String. ^bytes value-bytes "utf-8")))
 
 (defn add-encoded-cookie
-  "Inserts the provided name-value pair as a Cookie in the :cookies map of the response."
+  "Inserts the provided name-value pair as a Set-Cookie header in the response"
   [response password name value age-in-days]
   (letfn [(add-cookie-into-response [response]
-            (let [cookie-value {:value (encode-cookie value password), :max-age (-> age-in-days t/days t/in-seconds), :path "/"}]
-              (assoc-in response [:cookies name] cookie-value)))]
+            (let [encoded-cookie (-> (encode-cookie value password)
+                                     UrlEncoded/encodeString)
+                  max-age (-> age-in-days t/days t/in-seconds)
+                  path "/"
+                  set-cookie-header (str name "=" encoded-cookie ";Max-Age=" max-age ";Path=" path)
+                  existing-header (get-in response [:headers "set-cookie"])
+                  new-header (cond
+                               (nil? existing-header) set-cookie-header
```

Comment:
I'm on the fence about this. Jet and ring seem to use a string for a single-valued cookie and a vector for multiple values so I wanted to be consistent with them. I don't have a strong preference, though.